### PR TITLE
feat(GuildManager): Allow for more options for GuildManager.create

### PR DESCRIPTION
**Pull request by NotSugden**
**Please describe the changes this PR makes and why it should be merged:**
This PR allows for using all of the parameters shown [here](https://discordapp.com/developers/docs/resources/guild#create-guild) when creating a guild, (also changes Guild#verificationLevel and Guild#defaultMessageNotifications to use strings instead of numbers)

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by RDambrosio016** [comment]
Stores extend Collection, therefore the jsdoc must include the type of the key and value of the collection, so changing from `UserStore<Snowflake, User>` to `UserStore` is misleading and will lead to confusion.

**Comment by NotSugden** [comment]
> Stores extend Collection, therefore the jsdoc must include the type of the key and value of the collection, so changing from `UserStore<Snowflake, User>` to `UserStore` is misleading and will lead to confusion.

when i was testing, the jsdoc was being used instead of the typings file, `UserStore<Snowflake, User>` would show as an any type, while `UserStore` would show as a valid UserStore, its also refferred to as `UserStore` [here](https://github.com/discordjs/discord.js/blob/master/typings/index.d.ts#L1877) rather than `UserStore<Snowflake, User>`, so wouldnt `UserStore` be infact correct?

File: src/client/Client.js [hunk 1]
```diff
@@ -159,7 +159,7 @@ class Client extends BaseClient {
 
   /**
    * All custom emojis that the client has access to, mapped by their IDs
-   * @type {GuildEmojiStore<Snowflake, GuildEmoji>}
+   * @type {GuildEmojiStore}
```
Comment (NotSugden) [inline src/client/Client.js]:
```suggestion
   * @type {GuildEmojiStore<Snowflake, GuildEmoji>}
```

**Comment by NotSugden** [comment]
@SpaceEEC I believe thats everything (lmk if i missed anything)

File: src/managers/GuildManager.js [hunk 2]
```diff
@@ -36,6 +43,45 @@ class GuildManager extends BaseManager {
...
+   */
+
+  /**
+   * Partial data for a Channel.
+   * @typedef {Object} PartialChannelData
+   * @property {number} [id] The ID for this channel, used to set its parent,
+   * this is a placeholder and will be replaced by the API after consumption
+   * @property {number} [parentID] The parent ID for this channel
+   * @property {string} [type] The type of the channel
+   * @property {string} [name] The name of the channel
```
Comment (SpaceEEC) [inline src/managers/GuildManager.js]: The name is not optional.
```json
{
    "code": 50035,
    "errors": {
        "channels": {
            "0": {
                "name": {
                    "_errors": [
                        {
                            "code": "BASE_TYPE_REQUIRED",
                            "message": "This field is required"
                        }
                    ]
                }
            }
        }
    },
    "message": "Invalid Form Body"
}
```

```suggestion
   * @property {string} name The name of the channel
```

File: src/structures/Guild.js [hunk 3]
```diff
@@ -5,7 +5,8 @@ const Integration = require('./Integration');
 const GuildAuditLogs = require('./GuildAuditLogs');
 const Webhook = require('./Webhook');
 const VoiceRegion = require('./VoiceRegion');
-const { ChannelTypes, DefaultMessageNotifications, PartialTypes } = require('../util/Constants');
+const { ChannelTypes, DefaultMessageNotifications, PartialTypes, VerificationLevels,
+  ExplicitContentFilterLevels } = require('../util/Constants');
```
Comment (iCrawl) [inline src/structures/Guild.js]: Nit

**Pull request merged**